rpcs3/Emu/RSX/Common/BufferUtils.cpp

```diff
@@ -1033,7 +1037,7 @@ namespace
 	};
 
 	template<typename T>
-	std::tuple<T, T, u32> upload_untouched(gsl::span<to_be_t<const T>> src, gsl::span<T> dst, rsx::primitive_type draw_mode, bool is_primitive_restart_enabled, u32 primitive_restart_index)
+	std::tuple<T, T, u32> upload_untouched(gsl::span<to_be_t<const T>> src, gsl::span<T> dst, rsx::primitive_type draw_mode, bool is_primitive_restart_enabled, T primitive_restart_index)
```

Comment:
We really should keep restart index as u32 here and check if it is valid before dispatching the branch with restarts enabled.

Reply:
Fixed with an additional check, thanks.